src/all/kavita/src/eu/kanade/tachiyomi/extension/all/kavita/Kavita.kt

```diff
@@ -0,0 +1,1036 @@
+        screen.addPreference(
+            screen.editTextPreference(
+                ADDRESS_TITLE,
+                "",
+                "The OPDS url copied from User Settings. This should include address and the api key on end."
+            )
+        )
+    }
+
+    private fun androidx.preference.PreferenceScreen.editTextPreference(
+        title: String,
+        default: String,
+        summary: String,
+        isPassword: Boolean = false
+    ): androidx.preference.EditTextPreference {
+        return androidx.preference.EditTextPreference(context).apply {
+            key = title
+            this.title = title
+            val input = preferences.getString(title, null)
+            this.summary = if (input == null || input.isEmpty()) summary else input
+            this.setDefaultValue(default)
+            dialogTitle = title
+
+            if (isPassword) {
+                setOnBindEditTextListener {
+                    it.inputType =
+                        InputType.TYPE_CLASS_TEXT or InputType.TYPE_TEXT_VARIATION_PASSWORD
+                }
+            }
+            setOnPreferenceChangeListener { _, newValue ->
+                try {
+                    val res = preferences.edit().putString(title, newValue as String).commit()
+                    Toast.makeText(
+                        context,
+                        "Restart Tachiyomi to apply new setting.",
+                        Toast.LENGTH_LONG
+                    ).show()
+                    res
+                } catch (e: Exception) {
+                    e.printStackTrace()
+                    false
+                }
+            }
+        }
+    }
+
+    // private fun getPrefapiKey(): String = preferences.getString("APIKEY", "")!!
+    private fun getPrefBaseUrl(): String = preferences.getString("BASEURL", "")!!
+
+    // We strip the last slash since we will append it above
+    private fun getPrefAddress(): String {
+        var path = preferences.getString(ADDRESS_TITLE, "")!!
+        if (path.isNotEmpty() && path.last() == '/') {
+            path = path.substring(0, path.length - 1)
+        }
+        return path
+    }
+
+    companion object {
+        private const val ADDRESS_TITLE = "Address"
+        private val JSON_MEDIA_TYPE = "application/json; charset=utf-8".toMediaTypeOrNull()
+    }
+
+    /**
+     * LOGIN
+     **/
+    private fun setupLogin() {
+        val tokens = address.split("/opds/")
+        val apiKey = tokens[1]
+        val baseUrlSetup = tokens[0].replace("\n", "\\n")
+        if (!baseUrlSetup.startsWith("http")) {
+            throw Exception("""Url does not start with "http/s" but with ${baseUrlSetup.split("://")[0]} """)
+        }
+        preferences.edit().putString("APIKEY", apiKey).commit()
+        preferences.edit().putString("BASEURL", baseUrlSetup).commit()
+
+        val body = JSONObject().toString()
+            .toRequestBody("application/json; charset=utf-8".toMediaTypeOrNull())
```

Comment:
Can probably just hardcode the string and remove the usage of `org.json` stuff.

```suggestion
        val body = "{}".toRequestBody("application/json; charset=utf-8".toMediaTypeOrNull())
```